AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerServiceImpl.kt

```diff
@@ -196,16 +196,16 @@ internal class TestRunnerServiceImpl internal constructor(
             } else if (fileName.endsWith(SCREENSHOT_PNG_SUFFIX) || fileName.endsWith(SCREENSHOT_TEXT_PROTO_SUFFIX)) {
                 val runNumber = DeviceRun.create(visitor.fullDeviceId()).runNumber
                 // file names are in the format Classname_testname_emulator_goldResult.textproto or Classname_testname_emulator_actual/diff/expected.png
-                val className = fileName.split("_")[0]
-                val name = fileName.split("_emulator")[0].substringAfter("_")
+                val className = fileName.substringBefore('_')
+                val name = fileName.substringBefore("_emulator_").substringAfter('_')
```

Comment:
isn't this supported to be 
`fileName.substringAfter("_emulator_").substringBefore('_')` ?

what are we trying to extract exactly ?